### ISSUE TITLE
Flash Site playback

### ISSUE THREAD

**Issue by DJKOT2** [mozilla/shumway#2449]
Hi, I've made my portfolio site in Flash some time ago (using AS2).
Of course, now it's not working.
What should I do to make it work now? Do I have to convert it to HTML5?
How does your player works? Is it using some scripting to play normal regular SWF, or I have to convert SWF to something?

Please excuse my fairly illogical question, I don't know too much about programming.

--
Alex

**Comment by ComFreek** [comment]
> What should I do to make it work now? Do I have to convert it to HTML5?

Yes, you should. Either automated by the existing programs out there or manually. I am not too familiar with the internals of this project, but including another library on your site for just running Flash on-the-fly with JavaScript will slowdown your site. That's fine for nostalgic sites, but - in my opinion - not preferable for portfolio sites.

**Comment by ROBERT-MCDOWELL** [comment]
Flash is still very active and more powerful than html5/js, especially in the education sector, also js is hell to maintain with various browsers and version, lacking in very useful features like multicast, UDP, etc..
As a 20 years experienced developer who made js and as applications, I can say that the time passed to build a js app and debug was twice than as. I wanted to comment here to tell some truth about the reality since I'm very fed up to see so many disinformation about this subject.

**Comment by ROBERT-MCDOWELL** [comment]
Also you have a better alternative which is wasm, and there are already some AS to Wasm converters, search  on the net, you will find it.

**Comment by ROBERT-MCDOWELL** [comment]
@DJKOT2
why do you say it's not working? it should since Flash plugin has AVM1 and AVM2 runtime engines

**Comment by DJKOT2** [comment]
Flash is blocked or disabled by default in most browsers, so I determine that as "not working"

**Comment by ROBERT-MCDOWELL** [comment]
click on the your flash swf and a popup window will say "allow or block"

**Comment by DJKOT2** [comment]
I know that, but it's not the way for portfolio site to be done, to have a big "blocked" sign at the center
Some visitor of mine may not know how to unblock or enable the flash, so I can loose potential visitors, heck knows how much of them.
That's why I want either to convert my site to HTML5, or use something like Shumway

**Comment by ROBERT-MCDOWELL** [comment]
you under estimate users today, how do you embed your swf in your page? swfobject is the best way to do it so btw you can add a message like "please allow Flash to run by clicking below"
while you will reinvent the wheel by reprogramming everything in javascript. good luck

**Comment by DJKOT2** [comment]
I don't want everything to be reprogrammed. That's not the point.
Some of my clients using mobile phones to browse the web which won't allow them to see Flash content as far as I know. Youtube is okay, but Flash is no-no.
That's why I wanted to use Shumway but I don't know yet, how to do that.

**Comment by ROBERT-MCDOWELL** [comment]
youtube don't use flash at all, Flash is working well on smartphone with puffin browser.
do you really want to compete with my 20 years development expertise?

**Comment by DJKOT2** [comment]
Well, on my side I have 20 years of animation experience. Is it gonna change something? No.
I don't want to compete with you nor with anyone else. 
I want my Flash content to be probably played and viewed by any person who enters the site, without any special thingumabobs or big signs "please-please click that or that or that".

That's why I'm here.
I don't wanna argue about anything, I just wanna make things playable.

**Comment by ROBERT-MCDOWELL** [comment]
so you have to reprogram everything is javascrap

**Comment by ROBERT-MCDOWELL** [comment]
you have your experience, I have mine, and you are here to ask help about coding, not animation. Forget shumway, it's a dead project. btw javascript will also die soon... now do what you want, I tried to help you, but apparently it's not worth to do it.

**Comment by ROBERT-MCDOWELL** [comment]
and btw you should complain to the browser developers about your situation, not here

**Comment by DJKOT2** [comment]
I'm not quite sure browser developers are going to pay attention to what I say to them.
Judging by browsers getting slower and slower and dumber, I'm not expecting much help from browser devs.

**Comment by ROBERT-MCDOWELL** [comment]
well, you should learn why Flash is blocked, if it's not a browser decision. So they have to listen to you since your issue comes from Flash to be blocked. a browser plugin is managed by the browser, not the plugin itself

**Comment by DJKOT2** [comment]
maybe I should find halfway decent FLASH-HTML5 convertor?

**Comment by DJKOT2** [comment]
maybe I should find halfway decent FLASH-HTML5 convertor?

**Comment by ROBERT-MCDOWELL** [comment]
not for AS2 in my knowledge, and btw AS3 to html5/javascript is far far to be perfect, it can be ok (with bugs) for design, but for the code you will have to retouch it. html5 is not a substitute of Actionscript, you will have to dirt your fingers into javascript.

**Comment by DJKOT2** [comment]
even a simple scripts like "gotoandplay" or "geturl" won't work?

**Comment by ROBERT-MCDOWELL** [comment]
Lol!, no, it's call devolution :)

**Comment by SoniEx2** [comment]
maybe you should pay someone to code shumway.

**Comment by DJKOT2** [comment]
I can consider this as a possibility.
Is it true that it's having compatibility issues with latest Firefox?

**Comment by SoniEx2** [comment]
Yes, it doesn't work on latest Firefox.

**Comment by ROBERT-MCDOWELL** [comment]
you can check on my repo some new projects here, but just started
https://github.com/ROBERT-MCDOWELL/Flash

**Comment by DJKOT2** [comment]
To be honest, I cannot understand a word in there, as I'm not a developer.
It seems to me as a way of keeping old Flash alive, but I don't yet figured out how.

**Comment by ROBERT-MCDOWELL** [comment]
try to import your FLA into the last flash professional (animate CC as replaced name), then migrate your AS2 code into AS3, then save as HTML5/JS, hoping everything will be transcoded, but should be ok as it seems very basic code.

**Comment by DJKOT2** [comment]
I've checked these pages using firefox 62, working perfectly fine.
I can see all these demos and main page

https://mozilla.github.io/shumway/

https://mozilla.github.io/shumway/examples/inspector/inspector.html?rfile=../racing/race.swf

**Comment by TriMoon** [comment]
@ROBERT-MCDOWELL you should stop pushing your ideas on internet with arguments like "my 20 years of experience" because there are **a lot** of people on the internet that have way more experience as you...(I personally have 45+ years)

As for your comment "Forget shumway, it's a dead project." i think the community would agree more with a statement like "Forget Flash, it's prehistoric and doomed to disappear from usage".
Making something that provides a way to use OLD-TECH with NEW-TECH, like this projects intentions, should never be muffed away with arguments like yours...
Sure i can understand that you like to use that kind of arguments out of personal interest, but it is actually making you look worse because you will be admitting at same time that you are unable to adopt to new tech...

@DJKOT2 i am advicing the same as @ComFreek in https://github.com/mozilla/shumway/issues/2449#issuecomment-460921772
<s>You could look into using [Vue](https://vuejs.org/) for your portfolio, which will make you look better and allow you to use animations easily ;)</s>
I forgot you are not the coding type...

**Comment by DJKOT2** [comment]
@TriMoon
I'm afraid so.
But just of curiosity, what's Vue, and what it have to do with Flash portfoilo?
Is this some sort of framework? What's it doing?

**Comment by ROBERT-MCDOWELL** [comment]
@TriMoon
because for you new tech is javascript? give me a break.
also, my opinion is valid, if you don't like it just say you are not agree, but don't say to other you shouldn't say or do that, this is the basics of a democracy.
Sincerly.

**Comment by TriMoon** [comment]
@DJKOT2,
> @TriMoon
> I'm afraid so.
> But just of curiosity, what's Vue, and what it have to do with Flash portfoilo?
> Is this some sort of framework? What's it doing?

It's a framework to ease developing dynamic web-apps, you can find more info on its webpage i linked to :wink: 

@ROBERT-MCDOWELL,
> @TriMoon
> because for you new tech is javascript? give me a break.

you are clueless in IT that's a fact you have proven...
